src/components/Header.tsx

```diff
@@ -0,0 +1,54 @@
+import React, {ReactElement} from 'react';
+import {StyleProp, TextStyle, View} from 'react-native';
+import styles from '@styles/styles';
+import EnvironmentBadge from './EnvironmentBadge';
+import Text from './Text';
+
+type HeaderProps = {
+    /** Title of the Header */
+    title?: string | ReactElement;
+
+    /** Subtitle of the header */
+    subtitle?: string | ReactElement;
+
+    /** Should we show the environment badge (dev/stg)?  */
+    shouldShowEnvironmentBadge?: boolean;
+
+    /** Additional text styles */
+    textStyles?: StyleProp<TextStyle>;
+};
+
+function Header({title = '', subtitle = '', textStyles = [], shouldShowEnvironmentBadge = false}: HeaderProps) {
```

Comment:
Hoping maybe to learn something here, what's the term for what we're doing here where we have an object `{} : propTypes`? 

is this a typescript thing or a newer JS language thing?

Reply:
@bondydaa It is just the way of typing component props using TypeScript.
You can find similar example in the [TS_STYLE](https://github.com/Expensify/App/blob/main/contributingGuides/TS_STYLE.md#proptypes-and-defaultprops)

Reply:
👍 ah thanks, I'd completely forgot about "object destructing" addition to JS